File: 8.py
import math

def is_prime(n):
    if n < 2:
        return False
    for div in range(2, math.ceil(math.sqrt(n))+1):
        if n % div == 0:
            return False
    return True

num = int(input())
if is_prime(num):
    print("YES")
else:
    print("NO")

File: 8.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
int main()
{
    int num;
    scanf("%d", &num);

    if (num < 2)
        printf("NO\n");
    else
    {
        for (int i = 2; i < num; i++)
            if (num % i == 0)
            {
                printf("NO\n");
                exit(0);
            }
        printf("YES\n");
    }
}